common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -512,7 +514,9 @@ private static boolean eligibleToCache(CommandResult commandResult) {
      * @return
      */
     private static boolean eligibleToCacheException(BaseException exception) {
-        if (exception instanceof IntuneAppProtectionPolicyRequiredException) {
+        if (exception instanceof IntuneAppProtectionPolicyRequiredException
+                || (exception instanceof ClientException
+                && (ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())))) {
```

Comment:
```suggestion
                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())) {
```

Reply:
+1 on this feedback

Reply:
> +1 on this feedback

And the fact that it's ClientException is largely my fault, TTYTT.